site/source/docs/api_reference/wasm_workers.rst

```diff
@@ -0,0 +1,221 @@
+    <td class='cellborder'>Use the foo__proxy: 'sync'/'async' directive to specify a JS function to be run on the main thread context.</td>
+    <td class='cellborder'>N/A. JS code is always run on the calling thread context. Functions with __proxy directive will abort at runtime if called in a Worker.</td></tr>
+
+    <tr><td class='cellborder'>Proxied EM_ASM</td>
+    <td class='cellborder'>Use MAIN_THREAD_EM_ASM() and MAIN_THREAD_ASYNC_EM_ASM() to proxy EM_ASM code blocks to the main thread.</td>
+    <td class='cellborder'>N/A. Only calling thread EM_ASM() function blocks are possible.</td></tr>
+
+    <tr><td class='cellborder'>Build flags</td>
+    <td class='cellborder'>Compile and link with -pthread</td>
+    <td class='cellborder'>Compile and link with -sWASM_WORKERS=1</td></tr>
+
+    <tr><td class='cellborder'>Preprocessor directives</td>
+    <td class='cellborder'>__EMSCRIPTEN_PTHREADS__=1 and __EMSCRIPTEN_SHARED_MEMORY__=1 are active</td>
+    <td class='cellborder'>__EMSCRIPTEN_PTHREADS__=1, __EMSCRIPTEN_SHARED_MEMORY__=1 and __EMSCRIPTEN_WASM_WORKERS__=1 are active</td></tr>
+
+    <tr><td class='cellborder'>JS library directives</td>
+    <td class='cellborder'>USE_PTHREADS=1 and SHARED_MEMORY=1 are active</td>
+    <td class='cellborder'>USE_PTHREADS=1, SHARED_MEMORY=1 and WASM_WORKERS=1 are active</td></tr>
+
+    <tr><td class='cellborder'>Atomics API</td>
+    <td colspan=2>Supported, use any of <a href="https://gcc.gnu.org/onlinedocs/gcc/_005f_005fatomic-Builtins.html">__atomic_* API</a>, <a href="https://llvm.org/docs/Atomics.html#libcalls-sync">__sync_* API</a> or <a href="https://en.cppreference.com/w/cpp/atomic/atomic">C++11 std::atomic API</a>.</td></tr>
+
+    <tr><td class='cellborder'>Nonrecursive mutex</td>
+    <td class='cellborder'><pre>pthread_mutex_*</pre></td>
+    <td class='cellborder'><pre>emscripten_lock_*</pre></td></tr>
+
+    <tr><td class='cellborder'>Recursive mutex</td>
+    <td class='cellborder'><pre>pthread_mutex_*</pre></td>
+    <td class='cellborder'>N/A</td></tr>
+
+    <tr><td class='cellborder'>Semaphores</td>
+    <td class='cellborder'>N/A</td>
+    <td class='cellborder'><pre>emscripten_semaphore_*</pre></td></tr>
+
+    <tr><td class='cellborder'>Condition Variables</td>
+    <td class='cellborder'><pre>pthread_cond_*</pre></td>
+    <td class='cellborder'><pre>emscripten_condvar_*</pre></td></tr>
+
+    <tr><td class='cellborder'>Read-Write locks</td>
+    <td class='cellborder'><pre>pthread_rwlock_*</pre></td>
+    <td class='cellborder'>N/A</td></tr>
+
+    <tr><td class='cellborder'>Spinlocks</td>
+    <td class='cellborder'><pre>pthread_spin_*</pre></td>
+    <td class='cellborder'><pre>emscripten_lock_busyspin*</pre></td></tr>
+
+    <tr><td class='cellborder'>WebGL Offscreen Framebuffer</td>
+    <td class='cellborder'><pre>Supported with -sOFFSCREEN_FRAMEBUFFER=1</pre></td>
+    <td class='cellborder'><pre>Not supported.</pre></td></tr>
+
+  </table>
```

Comment:
Unfortunately it does. Trying to do RST's own tables resulted first in a ridiculous ASCII painting exercise, but then also the resulting table in HTML form did not flow properly on the page, and cut off really badly.

The HTML may be hard to read as a text file, but as a HTML output file it is what looks best, so I optimized for HTML output since that is what most users will read.